google/cloud/security/scanner/audit/instance_network_interface_rules_engine.py

```diff
@@ -0,0 +1,222 @@
+        resource_rules = self.rule_book.get_resource_rules()
+
+        for rule in resource_rules:
+            violations = itertools.chain(violations,
+                                         rule.find_violations(
+                                             instance_network_interface))
+        return violations
+
+    def add_rules(self, rules):
+        """Add rules to the rule book."""
+        if self.rule_book is not None:
+            self.rule_book.add_rules(rules)
+
+class InstanceNetworkInterfaceRuleBook(bre.BaseRuleBook):
+    """The RuleBook for enforced networks resources."""
+
+    def __init__(self,
+                 rule_defs=None):
+        """Initialize.
+
+        Args:
+            rule_defs: The parsed dictionary of rules from the YAML
+                definition file.
+            snapshot_timestamp: The snapshot to lookup data.
+        """
+        super(InstanceNetworkInterfaceRuleBook, self).__init__()
+        self.resource_rules_map = {}
+        if not rule_defs:
+            self.rule_defs = {}
+        else:
+            self.rule_defs = rule_defs
+            self.add_rules(rule_defs)
+
+    def add_rules(self, rule_defs):
+        """Add rules to the rule book."""
+        print(rule_defs)
+        for (i, rule) in enumerate(rule_defs.get('rules', [])):
+            self.add_rule(rule, i)
+
+    def add_rule(self, rule_def, rule_index):
+        """Add a rule to the rule book.
+
+        Add a rule to the rule book.
+
+        The rule supplied to this method is the dictionary parsed from
+        the rules definition file.
+
+        For example, this rule...
+
+            # rules yaml:
+            rules:
+              - project: a rule
+                network: whitelist
+                is_externam
+                whitelist:
+                  - project
+                        - network
+
+        ... gets parsed into:
+
+            {
+                'name': 'a rule',
+                'resource': {
+                    'project': '',
+                    'network': self,
+                    'is_external': boolean
+                }
+                whitelist {
+                    'network': ['project'...]
+
+                }
+            }
+
+        Args:
+            rule_def: A dictionary containing rule definition properties.
+            rule_index: The index of the rule from the rule definitions.
+                Assigned automatically when the rule book is built.
+
+        Args:
+            rule_def: A dictionary containing rule definition properties.
+            rule_index: The index of the rule from the rule definitions.
+            Assigned automatically when the rule book is built.
+        """
+        project = rule_def.get('project')
+        network = rule_def.get('network')
+        whitelist = rule_def.get('whitelist')
+        is_external_network = rule_def.get('is_external_network')
+
+        if (whitelist is None) or (project is None) or (network is None)\
```

Comment:
Readability: Please don't use \ for line continuations. Rather wrap the value in ().